galleon-plugins/src/main/java/org/wildfly/galleon/plugin/config/FileAppender.java

```diff
@@ -0,0 +1,154 @@
+
+/**
+ * Task to append content to the `target` file.
+ * The lines will be added to the first line of the target file that matches the `match` configuration.
+ * The lines to be added can be provided either via a file or via a list of lines.
+ * You can `ignore` the fact that the target file doesn't exist.
+ * @author Emmanuel Hugonnet
+ */
+public class FileAppender implements WildFlyPackageTask {
+
+    private String src;
+    private String target;
+    private String match;
+    private boolean ignore = true;
+    private final List<String> lines = new ArrayList<>();
+
+    public FileAppender() {
+    }
+
+    public String getTarget() {
+        return target;
+    }
+
+    public void setTarget(String target) {
+        this.target = target;
+    }
+
+    public Pattern getMatch() {
+        if(match == null) {
+            return null;
+        }
+        return Pattern.compile(match);
+    }
+
+    public void setMatch(String match) {
+        this.match = match;
+    }
+
+    public List<String> getLines() {
+        return Collections.unmodifiableList(lines);
+    }
+
+    public void addLine(String line) {
+        if(line == null) {
+            this.lines.add("");
+        } else {
+            this.lines.add(line);
+        }
+    }
+
+    public boolean isIgnore() {
+        return ignore;
+    }
+
+    public void setIgnore(boolean ignore) {
+        this.ignore = ignore;
+    }
+
+    public String getSource() {
+        return src;
+    }
+
+    public void setSource(String src) {
+        this.src = src;
+    }
+
+    @Override
+    public void execute(WfInstallPlugin plugin, PackageRuntime pkg) throws ProvisioningException {
+        if(this.getTarget() == null) {
+            if(this.isIgnore()) {
+                return;
+            }
+            throw new ProvisioningException("Target can't be null when appending content");
+        }
+        final Path target = plugin.getRuntime().getStagedDir().resolve(this.getTarget());
+        if (!Files.exists(target)) {
+            if(this.isIgnore()) {
+                return;
+            }
+            throw new ProvisioningException(Errors.pathDoesNotExist(target));
+        }
+        try {
+            final Path src;
+            List<String> linesToAppend;
+            if (this.getSource() != null) {
+                src = plugin.getRuntime().getStagedDir().resolve(this.getSource());
+                if (!Files.exists(src)) {
+                    throw new ProvisioningException(Errors.pathDoesNotExist(src));
+                }
+                linesToAppend = Files.readAllLines(src);
+            } else {
+                src = null;
+                linesToAppend = this.getLines();
+            }
+            List<String> lines = Files.readAllLines(target);
+            List<String> updatedLines = new ArrayList<>(linesToAppend.size() + 3);
+            for (String line : lines) {
+                if (this.getMatch() != null && this.getMatch().matcher(line).find()) {
```

Comment:
I'm just trying to clarify the use-cases you want to support here.
Are `linesToAppend` expected to be added multiple times in case the expression matches? Are there examples in our config where we would want that?
Is there an example where we want to append to the first line and then add the rest of the `linesToAppend` after that?
This functionality looks specialized for some specific use-cases, w/o knowing which this looks confusing.

Reply:
the initial need was for loging.properties: I need to add a logger to the loggers declaration (aka matching line) then add the logger configuration. We have this need for several subsystems thus avoiding the need to have all the loggers added in a specific file even if those loggers wouldn't be needed.

Reply:
I agree that once the match isd found and done we should just stop looking for it